### PR TITLE
Perf improvements

### PR DESCRIPTION
**Pull request by chris-smith-zocdoc**
@Carson-Carlisle-ZocDoc @daniel-meismer-zocdoc @david-chenevert-zocdoc  @eliah-hecht-zocdoc 

Improve the performance of `DataTypeConverter.Convert` for Enum types. 

The included test cases parses some values 100K times in a tight loop, I saw approx a 40-50% improvement.

Before
image

After
image

### CODE REVIEW

File: ZocUtil.Database.Tests/Reflection/ConvertHelperTests.cs [hunk 1]
```diff
@@ -1,116 +0,0 @@
-﻿using System;
```
Comment (chris-smith-zocdoc) [inline ZocUtil.Database.Tests/Reflection/ConvertHelperTests.cs]: I renamed this file, github is just confused.

File: ZocUtil.Database.Tests/Reflection/DataTypeConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
+﻿using System;
```
Comment (chris-smith-zocdoc) [inline ZocUtil.Database.Tests/Reflection/DataTypeConverterTests.cs]: renamed file

File: ZocUtil.Database/DataTypeConverter.cs [hunk 3]
```diff
@@ -112,14 +112,22 @@ public object Convert(object o, Type desiredType, string identifier = null)
             if (conversionType.IsEnum)
             {
                 //System.Convert.ChangeType doesn't work with enums
-                var str = o.ToString();
```
Comment (chris-smith-zocdoc) [inline ZocUtil.Database/DataTypeConverter.cs]: This string conversion is unnecessary if we already have an int/long/byte etc.

File: ZocUtil.Database/DataTypeConverter.cs [hunk 4]
```diff
@@ -112,14 +112,22 @@ public object Convert(object o, Type desiredType, string identifier = null)
...
+                // Enum.ToObject is the fastest conversion
+                if (o.GetType() != typeof (string))
+                {
+                    return Enum.ToObject(conversionType, o);
+                }
+
+                var str = (string) o;
 
                 long oAsLong;
                 if (long.TryParse(str, out oAsLong))
```
Comment (chris-smith-zocdoc) [inline ZocUtil.Database/DataTypeConverter.cs]: Left this here so you can parse int strings as enums, for whatever reason.. So in the test cases, its "50" -> 50 -> SomeEnum.Bar

**Comment by Carson-Carlisle-ZocDoc** [comment]
[LGTM](http://lgtm.in/i/SbYLNu5en)

**Comment by daniel-meismer-zocdoc** [comment]
:+1: 
Is there still a copy of this class in regular zocutil? If so you should make the change there too.

**Comment by chris-smith-zocdoc** [comment]
@daniel-meismer-zocdoc I don't think so, given that this class always shows up in my performance profiling :p

**Comment by kyle-fritz-zocdoc** [comment]
@daniel-meismer-zocdoc, no.  This class originated in proper ZocUtil and was moved here.  For a while, it existed in both places, but ZocUtil now calls this one.

**Pull request merged**